codespan/src/file.rs

```diff
@@ -2,68 +2,9 @@
 use serde::{Deserialize, Serialize};
 use std::ffi::{OsStr, OsString};
 use std::num::NonZeroU32;
-use std::{error, fmt};
 
 use crate::{ByteIndex, ColumnIndex, LineIndex, LineOffset, Location, RawIndex, Span};
-
-#[derive(Debug, PartialEq)]
-pub struct LineIndexOutOfBoundsError {
-    pub given: LineIndex,
-    pub max: LineIndex,
-}
-
-impl error::Error for LineIndexOutOfBoundsError {}
-
-impl fmt::Display for LineIndexOutOfBoundsError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(
-            f,
-            "Line index out of bounds - given: {}, max: {}",
-            self.given, self.max
-        )
-    }
-}
-
-#[derive(Debug, PartialEq)]
-pub enum LocationError {
-    OutOfBounds { given: ByteIndex, span: Span },
-    InvalidCharBoundary { given: ByteIndex },
-}
-
-impl error::Error for LocationError {}
-
-impl fmt::Display for LocationError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            LocationError::OutOfBounds { given, span } => write!(
-                f,
-                "Byte index out of bounds - given: {}, span: {}",
-                given, span
-            ),
-            LocationError::InvalidCharBoundary { given } => {
-                write!(f, "Byte index within character boundary - given: {}", given)
-            }
-        }
-    }
-}
-
-#[derive(Debug, PartialEq)]
-pub struct SpanOutOfBoundsError {
-    pub given: Span,
-    pub span: Span,
-}
-
-impl error::Error for SpanOutOfBoundsError {}
-
-impl fmt::Display for SpanOutOfBoundsError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(
-            f,
-            "Span out of bounds - given: {}, span: {}",
-            self.given, self.span
-        )
-    }
-}
+use codespan_reporting::files::Error;
```

Comment:
We'll need a `#[cfg(feature = "reporting")]` on this to get rid of the error - not sure how best to do this however, as the errors are needed for the method definitions! 😬 

Also as a nitpick, I like to group the external imports and crate imports separately.

Reply:
Adding that attribute will not really help because there are then issues of the Error type not being defined (cf. <https://github.com/brendanzab/codespan/pull/295/checks?check_run_id=1205871669#step:6:14>). As I said, this would make codespan dependent on codespan-reporting *non-optionally*.

> external imports and crate imports separately

Huh, thought rustfmt would take care of that already.

Reply:
I think either we give up the independent codespan crate or this is not possible. And since there are already plans of deprecating the codespan crate anyway, the decision seems quite easy for me.

Reply:
I don't particularly care either way, I don't really maintain saltwater anymore.

Reply:
I think I'll go ahead and remove that test then.